flaky test: TestBlockWait

```
2026-04-30T23:26:23.1100278Z test-1           | --- FAIL: TestBlockWait (6.63s)
2026-04-30T23:26:23.1100565Z test-1           |     client_test.go:474: 
2026-04-30T23:26:23.1100999Z test-1           |         	Error Trace:	/app/pkg/repl/client_test.go:474
2026-04-30T23:26:23.1101358Z test-1           |         	Error:      	Not equal: 
2026-04-30T23:26:23.1101649Z test-1           |         	            	expected: 0
2026-04-30T23:26:23.1101978Z test-1           |         	            	actual  : 1
2026-04-30T23:26:23.1102403Z test-1           |         	Test:       	TestBlockWait


2026-04-30T23:26:17.4197350Z test-1           | --- FAIL: TestBlockWait (7.90s)
2026-04-30T23:26:17.4197736Z test-1           |     client_test.go:474: 
2026-04-30T23:26:17.4198048Z test-1           |         	Error Trace:	/app/pkg/repl/client_test.go:474
2026-04-30T23:26:17.4198437Z test-1           |         	Error:      	Not equal: 
2026-04-30T23:26:17.4198756Z test-1           |         	            	expected: 0
2026-04-30T23:26:17.4199019Z test-1           |         	            	actual  : 2
2026-04-30T23:26:17.4199397Z test-1           |         	Test:       	TestBlockWait

```

--- a/pkg/repl/client.go
+++ b/pkg/repl/client.go
@@ -1028,6 +1028,7 @@ func (c *Client) BlockWait(ctx context.Context) error {
 
 	prevPos := c.getBufferedPos()
 	first := true
+	stallCount := 0
 	for {
 		select {
 		case <-ctx.Done():
@@ -1037,13 +1038,21 @@ func (c *Client) BlockWait(ctx context.Context) error {
 		default:
 			currPos := c.getBufferedPos()
 			if currPos.Compare(prevPos) <= 0 && !first {
-				// we skip flushing on the first iteration because c.getCurrentBinlogPosition already flushes the binary log
-
-				c.logger.Debug("buffered position has not advanced, flushing binary logs")
-				if err := dbconn.Exec(ctx, c.db, "FLUSH BINARY LOGS"); err != nil {
-					return err // it could be context cancelled, return it
+				// Position hasn't advanced. Only flush after multiple consecutive
+				// stalls to avoid unnecessary flushes when the binlog syncer is
+				// just slightly behind (e.g., under CI load). getCurrentBinlogPosition
+				// already flushes once at the start, so a brief stall is expected.
+				stallCount++
+				if stallCount >= 3 {
+					c.logger.Debug("buffered position has not advanced, flushing binary logs")
+					if err := dbconn.Exec(ctx, c.db, "FLUSH BINARY LOGS"); err != nil {
+						return err // it could be context cancelled, return it
+					}
+					c.flushedBinlogs.Add(1)
+					stallCount = 0
 				}
-				c.flushedBinlogs.Add(1)
+			} else {
+				stallCount = 0
 			}
 			prevPos = currPos
 			first = false

--- a/pkg/repl/client.go
+++ b/pkg/repl/client.go
@@ -54,6 +54,11 @@ const (
 	backoffMultiplier = 2
 	// Sleep time between position checks in BlockWait
 	blockWaitSleep = 100 * time.Millisecond
+	// Number of consecutive blockWaitSleep intervals where the buffered position
+	// hasn't advanced before BlockWait flushes binary logs to nudge the syncer.
+	// 3 * blockWaitSleep (~300ms) tolerates brief syncer lag (e.g. CI load) while
+	// remaining negligible relative to DefaultTimeout.
+	blockWaitStallThreshold = 3
 )
 
 var (
@@ -1043,7 +1048,7 @@ func (c *Client) BlockWait(ctx context.Context) error {
 				// just slightly behind (e.g., under CI load). getCurrentBinlogPosition
 				// already flushes once at the start, so a brief stall is expected.
 				stallCount++
-				if stallCount >= 3 {
+				if stallCount >= blockWaitStallThreshold {
 					c.logger.Debug("buffered position has not advanced, flushing binary logs")
 					if err := dbconn.Exec(ctx, c.db, "FLUSH BINARY LOGS"); err != nil {
 						return err // it could be context cancelled, return it

